File: Documents/trae_projects/lego_job/lego-mobile/src/api/__tests__/story.test.js
/**
 * Story API 单元测试
 */

import { storyAPI } from '../story';
import apiClient from '../client';

// Mock apiClient
jest.mock('../client', () => ({
  post: jest.fn(),
}));

describe('storyAPI', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('generate', () => {
    it('应该生成故事（带所有参数）', async () => {
      const mockResponse = { title: '新章节', content: '章节内容', puzzle: { question: '问题' } };
      apiClient.post.mockResolvedValue(mockResponse);

      const params = {
        characters: [{ character_id: '1', custom_name: '角色1' }],
        plot: '冒险',
        chapterCharacters: [{ character_id: '1', custom_name: '角色1' }],
        previousSummary: '前情提要',
        previousPuzzle: { question: '问题', answer: '答案', isCorrect: true },
        plotSelection: { weather: 'sunny', adventureType: 'exploration', terrain: 'forest', equipment: 'sword' },
        forcePuzzle: true,
      };

      const result = await storyAPI.generate(params);

      expect(apiClient.post).toHaveBeenCalledWith('/story', {
        characters: params.characters,
        plot: '冒险',
        chapterCharacters: params.chapterCharacters,
        previousSummary: '前情提要',
        previousPuzzle: params.previousPuzzle,
        plotSelection: params.plotSelection,
        forcePuzzle: true,
      });
      expect(result).toEqual(mockResponse);
    });

    it('应该生成故事（带最少参数）', async () => {
      const mockResponse = { title: '新章节', content: '章节内容' };
      apiClient.post.mockResolvedValue(mockResponse);

      const params = {
        characters: [{ character_id: '1', custom_name: '角色1' }],
        plot: '冒险',
      };

      const result = await storyAPI.generate(params);

      expect(apiClient.post).toHaveBeenCalledWith('/story', {
        characters: params.characters,
        plot: '冒险',
        chapterCharacters: undefined,
        previousSummary: undefined,
        previousPuzzle: undefined,
        plotSelection: undefined,
        forcePuzzle: undefined,
      });
      expect(result).toEqual(mockResponse);
    });
  });
});

File: Documents/trae_projects/lego_job/lego-mobile-android/data/src/test/java/com/legostory/mobile/data/model/StoryTest.kt
package com.legostory.mobile.data.model

import com.legostory.mobile.core.model.*
import org.junit.Test
import org.junit.Assert.*

class StoryTest {
    
    @Test
    fun `test PlotOption data class`() {
        val option = PlotOption(
            id = "weather-sunny",
            name = "晴天",
            icon = "☀️"
        )
        assertEquals("ID should match", "weather-sunny", option.id)
        assertEquals("Name should match", "晴天", option.name)
        assertEquals("Icon should match", "☀️", option.icon)
    }
    
    @Test
    fun `test PlotSelection default values`() {
        val selection = PlotSelection()
        assertNull("Weather should be null", selection.weather)
        assertNull("AdventureType should be null", selection.adventureType)
        assertNull("Terrain should be null", selection.terrain)
        assertNull("Equipment should be null", selection.equipment)
    }
    
    @Test
    fun `test PlotSelection with values`() {
        val selection = PlotSelection(
            weather = "sunny",
            adventureType = "adventure",
            terrain = "forest",
            equipment = "wand"
        )
        assertEquals("Weather should match", "sunny", selection.weather)
        assertEquals("AdventureType should match", "adventure", selection.adventureType)
        assertEquals("Terrain should match", "forest", selection.terrain)
        assertEquals("Equipment should match", "wand", selection.equipment)
    }
    
    @Test
    fun `test StoryCharacter data class`() {
        val character = StoryCharacter(
            characterId = "char-001",
            customName = "Hero",
            personality = "Brave",
            speakingStyle = "Formal"
        )
        assertEquals("CharacterId should match", "char-001", character.characterId)
        assertEquals("CustomName should match", "Hero", character.customName)
        assertEquals("Personality should match", "Brave", character.personality)
        assertEquals("SpeakingStyle should match", "Formal", character.speakingStyle)
    }
    
    @Test
    fun `test PreviousPuzzle data class`() {
        val puzzle = PreviousPuzzle(
            question = "What is the answer?",
            answer = "42",
            isCorrect = true
        )
        assertEquals("Question should match", "What is the answer?", puzzle.question)
        assertEquals("Answer should match", "42", puzzle.answer)
        assertTrue("IsCorrect should be true", puzzle.isCorrect)
    }
    
    @Test
    fun `test StoryGenerateRequest data class`() {
        val request = StoryGenerateRequest(
            characters = listOf(StoryCharacter("char-001", "Hero")),
            plot = "adventure",
            previousSummary = "Previous chapter summary",
            forcePuzzle = true
        )
        assertEquals("Characters should have 1 item", 1, request.characters.size)
        assertEquals("Plot should match", "adventure", request.plot)
        assertEquals("PreviousSummary should match", "Previous chapter summary", request.previousSummary)
        assertTrue("ForcePuzzle should be true", request.forcePuzzle!!)
    }
}
